jupiterone/client.py

```diff
@@ -349,3 +411,248 @@ def delete_relationship(self, relationship_id: str = None):
+
+    def create_integration_instance(self,
+                                    instance_name: str = None,
+                                    instance_description: str = None,
+                                    integration_definition_id: str = "8013680b-311a-4c2e-b53b-c8735fd97a5c"):
+        """Creates a new Custom Integration Instance.
+
+        args:
+            instance_name (str): The "Account name" for integration instance
+            instance_description (str): The "Description" for integration instance
+            integration_definition_id (str): The "Integration definition ID" for integration instance,
+            if no parameter is passed, then the Custom Integration definition ID will be used.
+        """
+        variables = {
+                  "instance": {
+                    "name": instance_name,
+                    "description": instance_description,
+                    "integrationDefinitionId": integration_definition_id,
+                    "pollingInterval": "DISABLED",
+                    "config": {
+                      "@tag": {
+                        "Production": False,
+                        "AccountName": True
+                      }
+                    },
+                    "pollingIntervalCronExpression": {},
+                    "ingestionSourcesOverrides": []
+                  }
+        }
+
+        response = self._execute_query(CREATE_INSTANCE, variables=variables)
+        return response['data']['createIntegrationInstance']
+
+    def fetch_all_entity_properties(self):
+        """Fetch list of aggregated property keys from all entities in the graph.
+
+        """
+
+        response = self._execute_query(query=ALL_PROPERTIES)
+
+        return_list = []
+
+        for i in response['data']['getAllAssetProperties']:
+
+            if i.startswith(('parameter.', 'tag.')) == False:
+
+                return_list.append(i)
+
+        return return_list
+
+    def fetch_all_entity_tags(self):
+        """Fetch list of aggregated property keys from all entities in the graph.
+
+        """
+
+        response = self._execute_query(query=ALL_PROPERTIES)
+
+        return_list = []
+
+        for i in response['data']['getAllAssetProperties']:
+
+            if i.startswith(('tag.')) == True:
+
+                return_list.append(i)
+
+        return return_list
+
+    def start_sync_job(self, instance_id: str = None):
+        """Start a synchronization job.
+
+        args:
+            instance_id (str): The "integrationInstanceId" request param for synchronization job
+        """
+        endpoint = "/persister/synchronization/jobs"
+
+        data = {
+               "source": "integration-managed",
+               "integrationInstanceId": instance_id
+               }
+
+        response = self._execute_syncapi_request(endpoint=endpoint, payload=data)
+
+        return response
+
+    def upload_entities_batch_json(self, instance_job_id: str = None, entities_list: list = None):
```

Comment:
Is it intentional that we're only implementing the `/entities` endpoint here and not including the ability to upload relationships? (either using the combined `/upload` or standalone `/relationships`)

Reply:
solid point, will review